roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -93,6 +104,10 @@ impl ParseClientJobDeclarationMessages for JobDeclaratorDownstream {
                 missing_txs.clone(),
             ));
 
+            if !txs_to_retrieve.is_empty() {
```

Comment:
Here you are asking the transactions to the rpc client. You are doing this _before_ sending `ProvideMissingTransactions` (PMT). My fear is that in this way the JD-server must wait the rpc client response in order to send `PMT` message. Please be careful!
If so, I suggest to save the variable `txs_to_retrieve` in the `declared_mining_job` field of `JobDeclaratorDownstream` and modify `add_tx_data_to_job` in order to be called in `main`.

Reply:
I don't think it's happening, since add_tx_data_to_job is not async and the handler doesn't await anything. Also analyzing logs it seems that PMT message is created and sent to JDC without waiting nothing

Reply:
Yes the point is is `add_tx_data_to_job` is not async. If it was, there wouldn't be any problem I think.